themes/bootstrap3/templates/RecordTab/holdingsils/extended.phtml

```diff
@@ -24,26 +25,16 @@
             <?=$this->transEsc('On Reserve - Ask at Circulation Desk')?><br>
           <?php endif; ?>
 
-          <?php if ($holding['use_unknown_message'] ?? false): ?>
+          <?php if ($availabilityStatus->useUnknownMessage()): ?>
             <span class="text-muted"><?=$this->transEsc('status_unknown_message')?></span>
           <?php else: ?>
-            <?php if ($holding['availability']): ?>
-              <?php /* Begin Available Items (Holds) */ ?>
-              <?php
-                if ((int)$holding['availability'] === \VuFind\ILS\Logic\ItemStatus::STATUS_UNCERTAIN) {
-                  $statusClassEsc = 'text-warning';
-                  $status = $holding['status'];
-                  $schemaAvailEsc = 'http://schema.org/LimitedAvailability';
-                } else {
-                  $statusClassEsc = 'text-success';
-                  $status = is_int($holding['availability']) && !empty($holding['status']) ? $holding['status'] : 'Available';
-                  $schemaAvailEsc = 'http://schema.org/InStock';
-                }
-              ?>
-              <span class="<?=$statusClassEsc?>"><?=$this->transEsc($status)?><link property="availability" href="<?=$schemaAvailEsc?>"></span>
-            <?php else: ?>
-              <?php /* Begin Unavailable Items (Recalls) */ ?>
-              <span class="text-danger"><?=$this->transEsc($holding['status'])?><link property="availability" href="http://schema.org/OutOfStock"></span>
+            <?php
+              $statusClassEsc = $this->availabilityStatus()->getClass($availabilityStatus);
+              $statusDescription = $availabilityStatus->getStatusDescription();
+              $schemaAvail = $availabilityStatus->getSchema();
+            ?>
+            <span class="<?=$statusClassEsc?>"><?=$this->transEsc($statusDescription)?><link property="availability" <?=$schemaAvail ? 'href="' . $this->escapeHtmlAttr($schemaAvail) . '"' : ''?>></span>
```

Comment:
I adjusted this slightly -- the `Esc` suffix on a variable means it is already escaped, so I changed `$schemaAvailEsc` to just `$schemaAvail` for accuracy.

However, I'm also wondering if we should skip the entire `<link>` tag when `$schemaAvail` is falsy. There's discussion in [VUFIND-1653](https://openlibraryfoundation.atlassian.net/browse/VUFIND-1653) about adding a mechanism to disable schema.org markup since it can be problematic in some situations; I wonder if this might be the first step toward implementing that.

What do you think?

Reply:
I am not very familiar with schema.org and what this link tag is used for. But having the link property without the `href` probably does not make much sense anyway.

Reply:
This is a mechanism for embedding linked data into HTML markup -- in this instance, the link tag associates an availability status with a URI representing that status. For this reason, I think it probably makes sense to rename the `getSchema` method to `getSchemaAvailabilityUri` to make this more understandable. I think the matter of disabling schema.org markup probably doesn't really impact this -- it's too widespread to handle in this layer and will likely need to be addressed by a separate view helper in another PR.

Reply:
Thanks for the explanation.

I changed the name of the method and wrapped an if statement around the whole link tag. This should make it easier to implement the proposed changes.